rules/rbe_repo.bzl

```diff
@@ -216,56 +214,74 @@ _VERBOSE = False
-        _print_exec_results("Resolve image digest", result, fail_on_error = True)
-        image_name = result.stdout.splitlines()[0]
-        print("Image with given tag `%s` is resolved to %s" %
-              (ctx.attr.tag, image_name))
+    # If not using checked-in configs we need to pull the container and resolve its tag to digest
+    # before we create the platform target.
+    if not ctx.attr.config_version:
+        # Perform some safety checks
+        _validate_host(ctx)
+        project_root = ctx.os.environ.get(_AUTOCONF_ROOT, None)
+
+        # TODO (nlopezgi): validate _AUTOCONF_ROOT points to a valid Bazel project
+        use_default_project = False
+        if not project_root:
+            if ctx.attr.output_base:
+                fail(("%s env variable must be set for rbe_autoconfig" +
+                      " to function properly when output_base is set") % _AUTOCONF_ROOT)
+
+            # TODO(nlopezgi): consider using native.existing_rules() to validate
+            # bazel_toolchains repo exists.
+            # Try to use the default project
+            # This is Bazel black magic, we're traversing the directories in the output_base,
+            # assuming that the bazel_toolchains external repo will exist in the
+            # expected path.
+            project_root = ctx.path(".").dirname.get_child("bazel_toolchains").get_child("rules").get_child("cc-sample-project")
+            if not project_root.exists:
+                fail(("Could not find default autoconf project in %s, please make sure " +
+                      "the bazel-toolchains repo is imported in your workspace with name " +
+                      "'bazel_toolchains' and imported before the rbe_autoconfig target " +
+                      "declaration ") % str(project_root))
+            project_root = str(project_root)
+            use_default_project = True
+
+        outputs_tar = ctx.attr.name + "_out.tar"
+
+        # Pull the image using 'docker pull'
+        _pull_image(ctx, image_name)
+
+        # If tag is specified instead of digest, resolve it to digest in the
+        # image_name as it will be used later on in the platform targets.
+        if ctx.attr.tag:
+            result = ctx.execute(["docker", "inspect", "--format={{index .RepoDigests 0}}", image_name])
+            _print_exec_results("Resolve image digest", result, fail_on_error = True)
+            image_name = result.stdout.splitlines()[0]
+            print("Image with given tag `%s` is resolved to %s" %
+                  (ctx.attr.tag, image_name))
+
+    # Create a default BUILD file with the platform + toolchain targets that
+    # will work with RBE with the produced toolchain
+    _create_platform(
+        ctx,
+        image_name = image_name,
+        name = name,
+    )
+
+    # If user picks rbe-ubuntu 16_04 container and
+    # a config exists for the current version of Bazel, create aliases and return
+    if ctx.attr.config_version:
```

Comment:
do this block at the top?

Reply:
I see why you call it here. _create_platform needs to run before it. It's bit weird to me that in the main logic you are doing a check on if ctx.attr.config_version, and in _create_platform you do it again.

Reply:
The check in _create_platform is done only to resolve the cc_toolchain_target. I could not think of a better way to structure this. I could resolve the cc_toolchain_target in this def and avoid the if in _create_platform, but that also seems odd and harder to maintain if there are changes to the platform targets.